### PR TITLE
feat: add a difficulty rating to each of the scenarios

### PR DESCRIPTION
**Pull request by gregoryduckworth**
### Description
Adds a difficulty rating to each of the scenarios. Rated out of 3 to signify Easy/Medium/Hard and a dropdown to filter by them.

### Checklist

- [x] I have read the contribution guidelines.
- [x] This PR includes tests (or updates existing tests) that cover the changes.
- [x] Documentation has been updated (if applicable).
- [x] Code follows the project's coding standards.
- [x] All tests pass.

### Screenshots (if applicable)
image
image

### CODE REVIEW

**Comment by e-taniguti** [comment]
overall this is looking really nice !!!!
Here's my comment:
Screenshot 2023-11-09 at 16 08 37

**Comment by e-taniguti** [comment]
Agreed.

The idea was to add the "Search" input in the Various Inputs scenario.
There is no input with this style and behaviour there (label and x icon).

**Comment by gregoryduckworth** [comment]
> Agreed.
> 
> The idea was to add the "Search" input in the Various Inputs scenario. There is no input with this style and behaviour there (label and x icon).

Yeah, good shout! I'll create an issue 👍

File: src/components/ScenarioBox.tsx [hunk 1]
```diff
@@ -28,18 +39,59 @@ const ScenarioBox: React.FC<ScenarioBoxProps> = ({
...
+          sx={{
+            overflow: 'hidden',
+            textOverflow: 'ellipsis',
+            flex: '1 0 auto',
+            display: 'flex',
+            justifyContent: 'flex-end',
+            alignItems: 'flex-end'
+          }}
+        >
+          <Typography variant='body2' paragraph>
```
Comment (e-taniguti) [inline src/components/ScenarioBox.tsx:91]: Its possible to replace the truncateDescription function by this css:
`    sx={{
            overflow: 'hidden',
            textOverflow: 'ellipsis',
            display: '-webkit-box',
            WebkitLineClamp: '2',
            WebkitBoxOrient: 'vertical'
          }}
`

Comment (e-taniguti) [reply src/components/ScenarioBox.tsx:91]: and we can also add the tooltip showing the entire description.

**Review by e-taniguti — APPROVED** [review]

**Pull request merged**